lib/vector-api-client/src/gql/tap.rs

```diff
@@ -0,0 +1,90 @@
+use crate::BoxedSubscription;
+use graphql_client::GraphQLQuery;
+
+/// Shorthand for a Chrono datetime, set to UTC.
+type DateTime = chrono::DateTime<chrono::Utc>;
+
+/// OutputLogEventsSubscription allows observability into the log events that are
+/// generated from component(s).
+#[derive(GraphQLQuery, Debug, Copy, Clone)]
+#[graphql(
+    schema_path = "graphql/schema.json",
+    query_path = "graphql/subscriptions/output_log_events.graphql",
+    response_derives = "Debug"
+)]
+pub struct OutputLogEventsSubscription;
+
+/// Tap encoding format type that is more convenient to use for public clients than the
+/// generated `output_log_events_subscription::LogEventEncodingType`.
+#[derive(Debug, Clone, Copy)]
+pub enum TapEncodingFormat {
+    Json,
+    Yaml,
+}
+
+/// String -> TapEncodingFormat, typically for parsing user input.
+impl std::str::FromStr for TapEncodingFormat {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "json" => Ok(Self::Json),
+            "yaml" => Ok(Self::Yaml),
+            _ => Err("Invalid encoding format".to_string()),
+        }
+    }
+}
+
+/// Map the public-facing `TapEncodingFormat` to the internal `LogEventEncodingType`.
+impl From<TapEncodingFormat> for output_log_events_subscription::LogEventEncodingType {
+    fn from(encoding: TapEncodingFormat) -> Self {
+        match encoding {
+            TapEncodingFormat::Json => Self::JSON,
+            TapEncodingFormat::Yaml => Self::YAML,
+        }
+    }
+}
+
+impl output_log_events_subscription::OutputLogEventsSubscriptionOutputLogEvents {
+    pub fn as_log(
+        &self,
+    ) -> Option<&output_log_events_subscription::OutputLogEventsSubscriptionOutputLogEventsOnLogEvent>
```

Comment:
What a type name! 🙂

Reply:
Tell me about it! Took some `cargo expand`ing to figure that one out.